docs/architecture/raft.rst

```diff
@@ -4,112 +4,250 @@ Raft Consensus Algorithm in ScyllaDB
 ==================================================
 
-Note that when you have a two-DC cluster with the same number of nodes in each DC, the cluster will lose the quorum if one 
+Note that when you have a two-DC cluster with the same number of nodes in each DC, the cluster will lose the quorum if one
 of the DCs is down.
 **We recommend configuring three DCs per cluster to ensure that the cluster remains available and operational when one DC is down.**
 
 Enabling Raft
 ---------------
 
-Enabling Raft in ScyllaDB 5.0
-===============================
+Enabling Raft in ScyllaDB 5.0 and 5.1
+=====================================
+
+.. warning::
+  In ScyllaDB 5.0 and 5.1, Raft is an experimental feature.
 
-.. note:: 
-  In ScyllaDB 5.0:
+It is not possible to enable Raft in an existing cluster in ScyllaDB 5.0 and 5.1.
+In order to have a Raft-enabled cluster in these versions, you must create a new cluster with Raft enabled from the start.
 
-  * Raft is an experimental feature.
-  * Raft implementation only covers safe schema changes. See :ref:`Safe Schema Changes with Raft <raft-schema-changes>`.
+.. warning::
+
+   **Do not** use Raft in production clusters in ScyllaDB 5.0 and 5.1. Such clusters won't be able to correctly upgrade to ScyllaDB 5.2.
 
-If you are creating a new cluster, add ``raft`` to the list of experimental features in your ``scylla.yaml`` file:
+   Use Raft only for testing and experimentation in clusters which can be thrown away.
+
+.. warning::
+    Once enabled, Raft cannot be disabled on your cluster. The cluster nodes will fail to restart if you remove the Raft feature.
+
+When creating a new cluster, add ``raft`` to the list of experimental features in your ``scylla.yaml`` file:
 
 .. code-block:: yaml
-    
+
     experimental_features:
      - raft
 
-If you upgrade to ScyllaDB 5.0 from an earlier version, perform a :doc:`rolling restart </operating-scylla/procedures/config-change/rolling-restart/>` 
-updating the ``scylla.yaml`` file for **each node** in the cluster to enable the experimental Raft feature:
+.. _enabling-raft-existing-cluster:
 
-.. code-block:: yaml
-    
-    experimental_features:
-      - raft
+Enabling Raft in ScyllaDB 5.2 and further
+=========================================
+
+.. TODO include enterprise versions in this documentation
 
+.. note::
+  In ScyllaDB 5.2, Raft is Generally Available and can be safely used for consistent schema management.
+  In ScyllaDB 5.3 it will become enabled by default.
+  In further versions it will be mandatory.
 
-When all the nodes in the cluster and updated and restarted, the cluster will begin to use Raft for schema changes.
+ScyllaDB 5.2 and later comes equipped with a procedure that can setup Raft in an existing cluster. We refer to this as the **internal Raft upgrade procedure** (do not confuse with the :doc:`ScyllaDB version upgrade procedure </upgrade/upgrade-opensource/upgrade-guide-from-5.1-to-5.2/upgrade-guide-from-5.1-to-5.2-generic>`).
```

Comment:
"Raft" is ambiguous, since we also plan to use it for strongly consistent tables.

We might say "Raft Cluster Metadata" everywhere. It's awkward but unambiguous.

Reply:
I'm pretty sure that strongly consistent tables won't be able to function without "Raft Cluster Metadata" - and since this upgrade procedure enables Raft-based Cluster Metadata, it in general enables Raft to work in Scylla (for any purpose).

Otherwise, I could say "internal Raft Cluster Metadata upgrade procedure", but is there a point? This is the place where the name of the procedure is first defined, there's no ambiguity, it's just a convenient name we pick to refer to it later.

Reply:
Changed:
> that can setup Raft in an existing cluster

to:
> that can setup Raft-based consistent cluster management in an existing cluster

but kept "internal Raft upgrade procedure"